src/CA2D.jl

```diff
@@ -24,86 +24,33 @@ function build_model(;rules::Tuple, dims=(100,100), Moore=true)
   return model
 end
 
-"""
-    periodic_neighbors(pos::Tuple{Integer, Integer}, dims::Tuple{Integer, Integer})
-
-Returns the the row and column numbers of the rows and columns before and after the pos.
-"""
-function periodic_neighbors(pos::Tuple{Integer, Integer}, dims::Tuple{Integer, Integer})
-  if pos[1] == 1
-    xbefore = dims[1]
-    if dims[1] == 1
-      xafter = 1
-    else
-      xafter = pos[1] + 1
-    end
-  elseif pos[1] == dims[1]
-    xbefore = pos[1] - 1
-    xafter = 1
-  else
-    xbefore = pos[1] - 1
-    xafter = pos[1] + 1
-  end
-  if pos[2] == 1
-    ybefore = dims[2]
-    if dims[2] == 1
-      yafter = 1
-    else
-      yafter = pos[2] + 1
-    end
-  elseif pos[2] == dims[2]
-    ybefore = pos[2] - 1
-    yafter = 1
-  else
-    ybefore = pos[2] - 1
-    yafter = pos[2] + 1
-  end
-  return (xbefore, ybefore), (xafter, yafter)
-end
-
 function ca_step!(model)
-  agentnum = nagents(model)
-  new_status = Array{String}(undef, agentnum)
-  for agid in 1:agentnum
-    agent = model.agents[agid]
-    coord = agent.pos
-    center = agent.status
-    before, after = periodic_neighbors(coord, model.space.dimensions)
-    right = model.agents[coord2vertex((after[1], coord[2]), model)].status
-    left = model.agents[coord2vertex((before[1], coord[2]), model)].status
-    top = model.agents[coord2vertex((coord[1], after[2]), model)].status
-    bottom = model.agents[coord2vertex((coord[1], after[2]), model)].status
-    topright = model.agents[coord2vertex((after[1], after[2]), model)].status
-    topleft = model.agents[coord2vertex((before[1], after[2]), model)].status
-    bottomright = model.agents[coord2vertex((after[1], before[2]), model)].status
-    bottomleft = model.agents[coord2vertex((before[1], before[2]), model)].status
-
-    if model.properties[:Moore]
-      nstatus = [topleft, top, topright, left, right, bottomleft, bottom, bottomright]
-      nlive = length(findall(a->a=="1", nstatus))
-    else
-      nstatus = [top, left, right, bottom]
-      nlive = length(findall(a->a=="1", nstatus))
+  new_status = Array{String}(undef, nagents(model))

```

Comment:
shouldn't this be `Bool` instead of `String`...? It takes up much less memory.

Reply:
How would you write a Bool array for the rules?

Reply:
By looking the code further down I see that `new_status` takes only the values `"0"` or `"1"`, so true or false?